pkg/messaging/broker_test.go

```diff
@@ -0,0 +1,250 @@
+package messaging
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	configv1alpha1 "github.com/openservicemesh/osm/pkg/apis/config/v1alpha1"
+
+	"github.com/openservicemesh/osm/pkg/announcements"
+	"github.com/openservicemesh/osm/pkg/k8s/events"
+)
+
+func TestAllEvents(t *testing.T) {
+	a := assert.New(t)
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+
+	c := NewBroker(stopCh)
+
+	proxyUpdateChan := c.GetProxyUpdatePubSub().Sub(announcements.ProxyUpdate.String())
+	defer c.Unsub(c.proxyUpdatePubSub, proxyUpdateChan)
+
+	podChan := c.GetKubeEventPubSub().Sub(
+		announcements.PodAdded.String(),
+		announcements.PodUpdated.String(),
+		announcements.PodDeleted.String(),
+	)
+	defer c.Unsub(c.kubeEventPubSub, podChan)
+
+	serviceChan := c.GetKubeEventPubSub().Sub(
+		announcements.ServiceAdded.String(),
+		announcements.ServiceUpdated.String(),
+		announcements.ServiceDeleted.String(),
+	)
+	defer c.Unsub(c.kubeEventPubSub, serviceChan)
+
+	meshCfgChan := c.GetKubeEventPubSub().Sub(announcements.MeshConfigUpdated.String())
+	defer c.Unsub(c.kubeEventPubSub, meshCfgChan)
+
+	certRotateChan := c.GetCertPubSub().Sub(announcements.CertificateRotated.String())
+	defer c.Unsub(c.certPubSub, certRotateChan)
+
+	numEventTriggers := 50
```

Comment:
Are we using 50 concurrent events per type to check if there is still deadlock?

Reply:
numEventTriggers corresponds to the number of times the test generates the batch of events (pod add|update|delete, service add|update|delete), and cert updates. It's just a test variable to trigger a reasonably large number of events.